vxquery-core/src/main/java/org/apache/vxquery/serializer/XMLSerializer.java

```diff
@@ -441,6 +446,34 @@ private void printSequence(PrintStream ps, SequencePointable seqp, String betwee
         }
     }
 
+    private void printArrayNode(PrintStream ps, TaggedValuePointable tvp) {
+        ArrayPointable ap = pp.takeOne(ArrayPointable.class);
+
+        try {
+            tvp.getValue(ap);
+            if (tvp.getTag() == ValueTag.ARRAY_TAG) {
```

Comment:
The TVP is an array. No need to check it in this function. Take a look at he other print functions.